Enable dist autograd tests

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28606 Enable dist autograd tests**

Without passing setup_model_parallel=True to @dist_init, the
decorator actually takes function object as the value for the
flag.

Differential Revision: [D18120507](https://our.internmc.facebook.com/intern/diff/D18120507)

mrshenli committed:
Enable dist autograd tests

Without passing setup_model_parallel=True to @dist_init, it the
decorator actually takes function object as the value for the
flag.

[ghstack-poisoned]

xush6528 commented:
I think we should assert `setup_model_parallel` is a bool to avoid this misuse going forward.

https://github.com/pytorch/pytorch/blob/59402f51cf393b5f3ce72fa2d3c99b98ed653afc/test/dist_utils.py#L54-L57

mrshenli commented:
Good point, let me add that.

mrshenli committed:
Update on "Enable dist autograd tests"


Without passing setup_model_parallel=True to @dist_init, the
decorator actually takes function object as the value for the
flag.

Differential Revision: [D18120507](https://our.internmc.facebook.com/intern/diff/D18120507)

[ghstack-poisoned]
